### ISSUE TITLE
DEPICT example error: Singular matrix

### ISSUE THREAD

**Issue by xztrr** [RebeccaFine/benchmarker#3]
Hi, thank you sharing this great program! I am going through the depict example code but when running the script depict_partitionHeritability.sh I got the following error message:

After merging with reference panel LD, 1162017 SNPs remain.
After merging with regression SNP LD, 1159186 SNPs remain.
Removed 54 SNPs with chi^2 > 457.824 (1159132 SNPs remain)
Traceback (most recent call last):
  File "ldsc/ldsc.py", line 644, in <module>
    sumstats.estimate_h2(args, log)
  File "ldsc/ldscore/sumstats.py", line 359, in estimate_h2
    twostep=args.two_step, old_weights=old_weights)
  File "ldsc/ldscore/regressions.py", line 346, in __init__
    slow=slow, step1_ii=step1_ii, old_weights=old_weights)
  File "ldsc/ldscore/regressions.py", line 208, in __init__
    jknife = jk.LstsqJackknifeFast(x, y, n_blocks)
  File "ldsc/ldscore/jackknife.py", line 310, in __init__
    self.delete_values = self.block_values_to_delete_values(xty, xtx)
  File "ldsc/ldscore/jackknife.py", line 424, in block_values_to_delete_values
    delete_xtx, delete_xty).reshape((1, p))
  File "anaconda2/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 403, in solve
    r = gufunc(a, b, signature=signature, extobj=extobj)
  File "anaconda2/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 97, in _raise_linalgerror_singular
    raise LinAlgError("Singular matrix")
LinAlgError: Singular matrix

I used the versions that were recommended in the environment.yml in the ldsc GitHub page:
- python=2.7
- bitarray=0.8
- nose=1.3
- numpy=1.12
- pandas=0.20
- scipy=0.18
- pybedtools=0.7

And I used the following files to run scripts:

**depict_annotateAndComputeLDScores.sh**

baseline_path=../../baseline_v1.1/baseline.${curr_chromosome}. (downloaded as 1000G_phase3_baseline_v1.1_ldscores.tgz from https://data.broadinstitute.org/alkesgroup/LDSCORE/)
gene_boundary_path=../data/GPL570ProbeENSGInfo+HGNC_reformatted_noMHC_RF.txt
bfile_path=../../1000G_EUR_Phase3_plink/1000G.EUR.QC.${curr_chromosome} (downloaded as 1000G_Phase3_plinkfiles.tgz)
print_snps_path=../../1000G_EUR_Phase3_baseline/print_snps.txt (downloaded as 1000G_Phase3_baseline_ldscores.tgz)

**depict_partitionHeritability.sh**

gwas_path=../../body_BMIz_forLDSC.sumstats.gz
baseline_path=../../baseline_v1.1/baseline.
gene_window_path=bmi_depict_50kb.
weights_path=../../1000G_Phase3_weights_hm3_no_MHC/weights.hm3_noMHC. (downloaded as 1000G_Phase3_weights_hm3_no_MHC.tgz)
frq_path=../../1000G_Phase3_frq/1000G.EUR.QC. (downloaded as 1000G_Phase3_frq.tgz)
ldsc_script_path=../../ldsc/ldsc.py
normalize_script_path=../src/add_stats_to_results_files.py \

Could it be that I downloaded a wrong file or that I am using a package in a wrong version?

**Comment by RebeccaFine** [comment]
Hi,

I'm not sure exactly what the error here is (and I'm not the developer of LDSC), but here's one guess. Did you check that the annotation you're testing has at least one SNP annotated to it on each chromosome? I don't believe LDSC functions properly unless the annotation exists on every chromosome (and, ideally, represents >=1% of the genome).

Best,
Rebecca

**Comment by xztrr** [comment]
Hi Rebecca, 

thank you for your help! I now checked the annotations and it has at least one SNP annotated to each chromosome. Maybe the error is then caused by LDSC if I have used the correct files?

**Comment by xztrr** [comment]
The error message is now an issue in the LDSC GitHub repository so I will follow that to find out if the error is caused by LDSC. 

I am now trying the MAGMA use case, but I couldn't find the file GPL570-GPL96-GPL1261-GPL1355TermGeneZScores-MGI_MF_CC_RT_IW_BP_KEGG_z_z_GTExGenesOnly.txt needed for 3_gene_set_enrichment_analysis.sh that should be in the data folder. Could you add this file or is it located somewhere else?

**Comment by RebeccaFine** [comment]
Good luck! You could also try posting in their Google Group. Maybe check that the heritability partitioning works with just the baseline model? That would at least tell you if the issue is with your annotations and not somewhere in the code.

The file you're looking for is on my FTP site: ftp://ftp.broadinstitute.org/outgoing/benchmarker_data/.

**Comment by xztrr** [comment]
Thank you so much!
I tried the heritability partitioning with just the baseline model and it worked so the issue seems to be with my annotations. I now checked again the expected output scripts of the depict example and there the gene_window_path is set to ../gene_window_files/depict_and_gtex_gene_intersection_50kb_geneWindow. . I accidentally put in my script the path to the annotation generated by depict_annotateAndComputeLDScores.sh. I now re-ran the script with the correct file (downloaded from  ftp://ftp.broadinstitute.org/outgoing/benchmarker_data/) and it worked! I'm now wondering if these geneWindow files were obtained with the scripts in the generate_gene_window folder, because I want to try to run benchmarker on some other GWAS data. 
Can I just run 1_annotate_geneWindow with 1000G_EUR_Phase3_baseline/baseline.%s.annot.gz as baseline file and GPL570ProbeENSGInfo+HGNC_reformatted_noMHC_depictAndGtexGeneIntersection_RF.txt as --gene-boundary file and then 2_compute_LD_scores_geneWindow.sh with --bfile 1000G_EUR_Phase3_plink/1000G.EUR.QC. and --print-snps 1000G_EUR_Phase3_baseline/print_snps.txt and that would then be the correct gene window file for an analysis with European 1000G as reference?

**Comment by RebeccaFine** [comment]
I've updated the gene window processing steps to be simplified and easier to run -- this should answer your questions, as the files are listed in the annotate_and_compute_ld_scores_geneWindow_template.sh I've added to the src directory. The only immediate difference I noted between the files I used and those you mentioned are that I used the baseline_v1.1/ annotations rather than the baseline/ ones.

Let me know if you have further questions!
